### DISCUSSION TITLE
Unsatisfied dependency: no bean matches the injection point

### DISCUSSION THREAD

**Discussion by softshipper** [quarkusio/quarkus#25120] [Q&A] [answered]
Hi all

I use the Kotlin language to develop Quarkus based application. 
Regarding to the https://quarkus.io/guides/kotlin it is possible to write constructor injection in the class definition.

I am trying to do so and receive a warning:

[![enter image description here][1]][1]

What am I doing wrong?

The project is hosted on https://github.com/softshipper/playwithmtls.

Best regards

  [1]: https://i.stack.imgur.com/6FwRE.png

[a comment by an automated account is not shown]

**Comment by mkouba** [comment, marked as answer]
@softshipper This warning comes from the IDE and not quarkus itself, right?

**Reply by softshipper** [reply]
Yes, exactly. It comes from IntellJ

**Reply by mkouba** [reply]
In that case, it's up to the IntellJ to determine the correct set of beans in your application. It's very likely that it does not see all the resolved dependencies.

**Reply by softshipper** [reply]
I have added the dependency `implementation("io.quarkus:quarkus-smallrye-jwt")` and the warning disappears.
Thanks
